frontend/cypress/tests/codeView/codeView_switchAuthorship.js

```diff
@@ -0,0 +1,69 @@
+describe('switch authorship', () => {
+  it('switch authorship view should restore all default controls', () => {
+    Cypress.wait();
+
+    // open the code panel
+    cy.get('.icon-button.fa-code')
+        .should('be.visible')
+        .first()
+        .click();
+
+    // switch authorship view
+    cy.get('.icon-button.fa-code')
```

Comment:
Hello, I have a question regarding this. Will it be necessary to have a check on the sorting and file types before the switch happens?

Reply:
This test fundamentally checks if default controls are restored after switching between authorship tabs, so I don't see a need to perform checks before the switch. However, if the first authorship view is regarded as "switching of authorship view", then it can be included.

Reply:
Yeah that makes sense. However, I am wondering whether it will be necessary to check if previous authorship tab control can affect the control after the switch. Will it be necessary to alter some of the tab control setting first before making the switch, and then check if the default control is restored? Still, it is just some personal thought.

Reply:
Thanks for the input @HCY123902, altering the tab control settings before the switch is a valid consideration and has been implemented.